### PR TITLE
Add a helper function that returns site ID and site URL in a `Site` struct

### PR DESCRIPTION
**Pull request by aartij17**
This PR adds a helper function that returns a newly introduced `Site` struct that has a site ID and site webURL for a given m365 tenant.

---

#### Does this PR need a docs update or release note?
- [X] :no_entry: No

#### Type of change

- [X] :sunflower: Feature

#### Issue(s)

* #2777 

#### Test Plan

- [X] :zap: Unit test

```
--- PASS: TestM365IntegrationSuite (2.49s)
    --- PASS: TestM365IntegrationSuite/TestSites (1.82s)
        --- PASS: TestM365IntegrationSuite/TestSites/site (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#01 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#02 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#03 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#04 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#05 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#06 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#07 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#08 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#09 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#10 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#11 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#12 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#13 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#14 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#15 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#16 (0.00s)
        --- PASS: TestM365IntegrationSuite/TestSites/site#17 (0.00s)
```

### CODE REVIEW

**Review by enothereska — APPROVED** [review]

File: src/pkg/services/m365/m365_test.go [hunk 1]
```diff
@@ -48,3 +48,26 @@ func (suite *M365IntegrationSuite) TestUsers() {
...
+
+	var (
+		t    = suite.T()
+		acct = tester.NewM365Account(suite.T())
+	)
+
+	sites, err := Sites(ctx, acct, fault.New(true))
+	require.NoError(t, err)
+	require.NotNil(t, sites)
+	require.Greater(t, len(sites), 0)
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/m365_test.go:64]: Nits: 
* `require.` should only be used if the test cannot run beyond this failure.  This should be an assert.
* the order of values should be `assert.Foo(t, expected, result)`.  So in this case: `assert.Less(t, 0, len(sites)).
* Greater/Less is indirect; you probably want `NotEmpty` instead.
```suggestion
	assert.NotEmpty(t, sites)
```

File: src/pkg/services/m365/m365.go [hunk 2]
```diff
@@ -86,6 +86,34 @@ func UserPNs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]stri
 	return ret, nil
 }
 
+type Site struct {
+	// URL that displays the item in the browser
+	URL string
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/m365.go:91]: nit: I'm being overly picky about naming consistency.
```suggestion
	WebURL string
```

File: src/pkg/services/m365/m365.go [hunk 3]
```diff
@@ -86,6 +86,34 @@ func UserPNs(ctx context.Context, acct account.Account, errs *fault.Bus) ([]stri
 	return ret, nil
 }
 
+type Site struct {
+	// URL that displays the item in the browser
```
Comment (ryanfkeepers) [inline src/pkg/services/m365/m365.go:90]:
```suggestion
	// WebURL identifying the site.  Ex: https://contoso.sharepoint.com/teams/hr
```

**Review by ryanfkeepers — APPROVED** [review]
Looks good, just some nits.

File: src/pkg/services/m365/m365.go (hunk 2 above)
Comment (aartij17) [reply src/pkg/services/m365/m365.go:91]: I appreciate nits always! I will change it, thank you.

File: src/pkg/services/m365/m365_test.go (hunk 1 above)
Comment (aartij17) [reply src/pkg/services/m365/m365_test.go]: Done. Thank you!

**Pull request merged**